plugins/org.eclipse.n4js/src/org/eclipse/n4js/utils/WildcardPathFilterHelper.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2018 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.utils;
+
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.n4js.internal.N4JSModel;
+import org.eclipse.n4js.internal.N4JSProject;
+import org.eclipse.n4js.n4mf.ModuleFilter;
+import org.eclipse.n4js.n4mf.ModuleFilterSpecifier;
+import org.eclipse.n4js.projectModel.IN4JSCore;
+import org.eclipse.n4js.projectModel.IN4JSSourceContainer;
+
+import com.google.common.base.Preconditions;
+import com.google.inject.Inject;
+
+/**
+ * Utility methods for wildcards in project description e.g. in module filters
+ */
+public class WildcardPathFilterHelper {
+
+	@Inject
+	private IN4JSCore n4jsCore;
+
+	@Inject
+	private N4JSModel n4jsModel;
+
+	/** @return true iff the given location is matched the given filter */
+	public boolean isPathContainedByFilter(URI location, ModuleFilter filter) {
+		for (ModuleFilterSpecifier spec : filter.getModuleSpecifiers()) {
+			String prjRelativeLocation = getProjectRelativeLocation(location, spec);
+			if (prjRelativeLocation != null) {
+				boolean isContained = locationMatchesGlobSpecifier(spec, prjRelativeLocation);
+				if (isContained) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/** @return true iff the given location is matched the given specifier */
+	public boolean isPathContainedByFilter(URI location, ModuleFilterSpecifier spec) {
+		String prjRelativeLocation = getProjectRelativeLocation(location, spec);
+		if (prjRelativeLocation != null) {
+			return locationMatchesGlobSpecifier(spec, prjRelativeLocation);
+		}
+		return false;
+	}
+
+	private String getProjectRelativeLocation(URI location, ModuleFilterSpecifier spec) {
+		IN4JSSourceContainer sourceContainer = n4jsCore.findN4JSSourceContainer(location).orNull();
+		if (sourceContainer == null) {
+			return null;
+		}
+
+		Path locationPath = Paths.get(location.toString());
+		String filterSrcCont = spec.getSourcePath();
+		if (filterSrcCont == null || filterSrcCont.equals(sourceContainer.getRelativeLocation())) {
```

Comment:
shouldn't check for the module wrapper be done earlier, and not be considered in this method at all?